### ISSUE TITLE
Seeking an alternative to Planetscale for local development

### ISSUE THREAD

**Issue by harshalmittal4** [dubinc/dub#596]
hello, 
I wish to contribuet to this project, but while settign it up locally following [the guide](https://dub.co/docs/local-development), I discovered that PlanetScale free plan is not available in my location. I was wondering if there is an alternative to PlanetScale that is recommended. Thanks!

![Screenshot 2024-02-03 at 8 00 21 PM](https://github.com/dubinc/dub/assets/24206326/7513ba9b-a767-4ab1-a085-9f22d85dd8fd)

**Comment by versecafe** [comment]
@harshalmittal4 you should be able to run any MySQL DB locally or on a service like railway and use a connection string, if you have any issues with that just let me know.

**Comment by steven-tey** [comment]
@versecafe that's true but one caveat is you won't be able to do some of the stuff here: https://github.com/dubinc/dub/blob/main/apps/web/lib/planetscale.ts

**Comment by harshalmittal4** [comment]
Thanks @versecafe and @steven-tey ! I have not used railway earlier but will try it.

**Comment by steven-tey** [comment]
We updated our self-hosting & local dev guides recently to reflect updates from Planetscale's side: https://twitter.com/steventey/status/1767196326549442656

**Comment by steven-tey** [comment]
the planetescale docker simulator solves the limitation we discussed above as well btw @versecafe

**Comment by rfrlcode** [comment]
I believe I might have found a way to replace planetscale with a different mysql provider, while keeping the features from planetscale.ts working completely.

First, startup a hosted mysql instance somewhere. Can be anywhere. I chose aiven.io.
Get the mysql uri from them. it will look like this:
mysql://username:[password@hostname.a.aivencloud.com](mailto:password@hostname.a.aivencloud.com):28146/dbname?ssl-mode=REQUIRED

add this to your .env as the DATABASE_URL

We will now work on a way to deploy the docker image for the planetscale http sim.
Create a folder anywhere (can be named anything). This folder should contain two files:

docker-compose.yml
version: "3"
services:
planetscale-proxy:
build:
context: .
dockerfile: Dockerfile
environment:

MYSQL_DBNAME=dbname
MYSQL_ADDR=hostname.a.aivencloud.com
MYSQL_PORT=28146
Dockerfile
FROM ghcr.io/mattrobenolt/ps-http-sim:latest
CMD ["-listen-port=3900", "-mysql-dbname=dbname", "-mysql-addr=hostname.a.aivencloud.com", "-mysql-port=28146"]

Deploy this docker image anywhere. I used render.com for free hosting (upload the above folder with two files to github, and then deploy to render using github). render will give you a url like this: https://examples.onrender.com/

Use the url, to add the following to your dub .env file (the username and password come from the mysql uri above)
PLANETSCALE_DATABASE_URL="https://username:password@examples.onrender.com/"

now pnpm run dev, and everything should work as usual, and we dont have to pay the $$ for Planetscale.
Let me know what yall think, and if something can be improved!
